tests/test_tmpdir.py

```diff
@@ -285,6 +287,216 @@ def test_dockerfile_singularity_build(monkeypatch: pytest.MonkeyPatch, tmp_path:
+    repo_path.mkdir()
+    image_path = repo_path / "alpine"
+    image_path.mkdir()
+
+    runtime_context = RuntimeContext({})
+    builder = Builder(
+        {},
+        [],
+        [],
+        {},
+        schema.Names(),
+        [],
+        [],
+        {},
+        None,
+        None,
+        StdFsAccess,
+        StdFsAccess(""),
+        None,
+        0.1,
+        True,
+        False,
+        False,
+        "no_listing",
+        runtime_context.get_outdir(),
+        runtime_context.get_tmpdir(),
+        runtime_context.get_stagedir(),
+        INTERNAL_VERSION,
+        "singularity",
+    )
+
+    monkeypatch.setattr(
+        "cwltool.singularity._inspect_singularity_image", lambda *args, **kwargs: True
+    )
+    # ensure then env var is not set
+    if os.environ.get("CWL_SINGULARITY_IMAGES", None):
+        os.environ.pop("CWL_SINGULARITY_IMAGES")
+    initial_requirements = {"class": "DockerRequirement", "dockerPull": "alpine"}
+    requirements = copy.deepcopy(initial_requirements)
+    # get image from sandbox_base_path option
+    res_get_image = SingularityCommandLineJob(
+        builder, {}, CommandLineTool.make_path_mapper, [], [], ""
+    ).get_image(
+        requirements,
+        pull_image=False,
+        tmp_outdir_prefix=str(tmp_outdir_prefix),
+        force_pull=False,
+        sandbox_base_path=str(repo_path.absolute()),
+    )
+    assert res_get_image
+    assert requirements["dockerImageId"] == "alpine"
+
+    requirements = copy.deepcopy(initial_requirements)
+    requirements_obj: CWLObjectType = {"class": "DockerRequirement", "dockerPull": "alpine"}
+
+    res_get_req = SingularityCommandLineJob(
+        builder, {}, CommandLineTool.make_path_mapper, [], [], ""
+    ).get_from_requirements(
+        requirements_obj,
+        pull_image=False,
+        force_pull=False,
+        tmp_outdir_prefix=str(tmp_outdir_prefix),
+        image_base_path=str(repo_path.absolute()),
+    )
+    assert res_get_req == str(os.path.abspath(image_path))
+
+    requirements = copy.deepcopy(initial_requirements)
+    # get requirements from without sandbox image path
+    # should return an error
+    with pytest.raises((WorkflowException, KeyError)):
+        res_get_req = SingularityCommandLineJob(
+            builder, {}, CommandLineTool.make_path_mapper, [], [], ""
+        ).get_from_requirements(
+            requirements_obj,
+            pull_image=False,
+            force_pull=False,
+            tmp_outdir_prefix=str(tmp_outdir_prefix),
+        )
+
+    # get image from CWL_SINGULARITY_IMAGES env var
+    requirements = copy.deepcopy(initial_requirements)
+    os.environ["CWL_SINGULARITY_IMAGES"] = str(repo_path.absolute())
+    res_get_image = SingularityCommandLineJob(
+        builder, {}, CommandLineTool.make_path_mapper, [], [], ""
+    ).get_image(
+        requirements,
+        pull_image=False,
+        tmp_outdir_prefix=str(tmp_outdir_prefix),
+        force_pull=False,
+    )
```

Comment:
Please use a monkeypatch to adjust the environ, needed for parallel execution of tests
https://docs.pytest.org/en/stable/how-to/monkeypatch.html#monkeypatching-environment-variables